From d45a483b4fce68979d24ec660a2bec5a9dad1b0b Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:20 +0100
Subject: [PATCH 36/51] Revert "Buffer the pending instantiations in case or
 recursive parsing."

This reverts commit 7d9a6f43951b5a0d154160dc3452a4c09a9b0769.
---
 .../clang/include/clang/Sema/Sema.h            | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/interpreter/llvm-project/clang/include/clang/Sema/Sema.h b/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
index 12a0ec36c4..e7ed553ec5 100644
--- a/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
+++ b/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
@@ -10274,24 +10274,6 @@ public:
     }
   };
 
-  class SavePendingInstantiationsRAII {
-  public:
-    SavePendingInstantiationsRAII(Sema &S):
-      SavedPendingLocalImplicitInstantiations(S), S(S) {
-      SavedPendingInstantiations.swap(S.PendingInstantiations);
-    }
-
-    ~SavePendingInstantiationsRAII() {
-      assert(S.PendingInstantiations.empty() &&
-             "there shouldn't be any pending instantiations");
-      SavedPendingInstantiations.swap(S.PendingInstantiations);
-    }
-  private:
-    LocalEagerInstantiationScope SavedPendingLocalImplicitInstantiations;
-    Sema &S;
-    std::deque<PendingImplicitInstantiation> SavedPendingInstantiations;
-  };
-
   void PerformPendingInstantiations(bool LocalOnly = false);
 
   TypeSourceInfo *SubstType(TypeSourceInfo *T,
-- 
2.43.5

